scripts/evaluate_similarity.py

```diff
@@ -18,19 +18,33 @@ def parse_args(args):
         help="Similarity algorithm to use",
         choices=["lsi", "neighbors_tfidf", "neighbors_tfidf_bigrams", "word2vec_wmd"],
     )
+    parser.add_argument(
+        "--disable-url-cleanup",
+        help="Don't cleanup urls when training the similarity model",
+        dest="cleanup_urls",
+        default=True,
+        action="store_false",
+    )
     return parser.parse_args(args)
 
 
 def main(args):
     if args.algorithm == "lsi":
-        model = LSISimilarity()
+        model_creator = LSISimilarity
     elif args.algorithm == "neighbors_tfidf":
-        model = NeighborsSimilarity()
+        model_creator = NeighborsSimilarity
     elif args.algorithm == "neighbors_tfidf_bigrams":
-        model = NeighborsSimilarity(vectorizer=TfidfVectorizer(ngram_range=(1, 2)))
+        model_creator = NeighborsSimilarity
     elif args.algorithm == "word2vec_wmd":
-        model = Word2VecWmdSimilarity()
-
+        model_creator = Word2VecWmdSimilarity
+
+    if args.algorithm == "neighbors_tfidf_bigrams":
+        model = model_creator(
```

Comment:
This kind of defies the purpose of having a common model_creator.
In the `neighbors_tfidf_bigrams` above, you can define model_creator as a function that calls NeighborsSimilarity with the vectorizer argument + other arguments passed to the function.